lib/lists.nix

```diff
@@ -57,73 +81,141 @@ rec {
             else [ (f (n + 1) (elemAt list n)) ] ++ imap' (n + 1);
       in imap' 0;
 
+  /* Map and concatenate the result.
 
-  # Map and concatenate the result.
+     Example:
+       concatMap (x: x ++ ["z"]) ["a" "b"]
+       => [ "a" "z" "b" "z" ]
```

Comment:
nix-repl> concatMap (x: x ++ ["z"]) ["a" "b"]
error: value is a string while a list was expected, at (string):1:15